### PR TITLE
Apps refactor

### PR DESCRIPTION
**Pull request by freesig**
Rewrite app utils and token to use builder and node

### CODE REVIEW

File: pkgs/essential-all.nix
Comment (supiket) [inline pkgs/essential-all.nix:1]: Nice

File: apps/utils/src/lib.rs [hunk 1]
```diff
@@ -1,9 +1,11 @@
 pub mod addresses;
-pub mod cli;
 pub mod compile;
-#[cfg(feature = "test-utils")]
-pub mod debug;
+// #[cfg(feature = "test-utils")]
+// pub mod debug;
```
Comment (supiket) [inline apps/utils/src/lib.rs:4]:
```suggestion
```

**Review by mitchmindtree — APPROVED** [review]
Had a quick skim :shipit:

File: apps/utils/src/db.rs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+pub async fn new_dbs() -> Dbs {
+    let config = node::db::Config {
+        source: node::db::Source::Memory(uuid::Uuid::new_v4().to_string()),
+        ..Default::default()
+    };
+    // let config = node::db::Config {
+    //     source: node::db::Source::Path(concat!(env!("CARGO_MANIFEST_DIR"), "/db.db").into()),
+    //     ..Default::default()
+    // };
```
Comment (supiket) [inline apps/utils/src/db.rs:17]:
```suggestion
```

File: apps/counter/app/Cargo.toml [hunk 3]
```diff
@@ -17,4 +16,7 @@ tokio.workspace = true
 
 [dev-dependencies]
 essential-app-utils = { workspace = true, features = ["test-utils"] }
-essential-wallet = { workspace = true, features = ["test-utils"] }
\ No newline at end of file
+essential-builder.workspace = true
+essential-builder-db.workspace = true
+essential-node.workspace = true
+essential-node-db = { workspace = true }
```
Comment (supiket) [inline apps/counter/app/Cargo.toml:22]:
```suggestion
```

File: apps/counter/app/tests/tests.rs [hunk 4]
```diff
@@ -17,33 +20,71 @@ async fn number_go_up() {
...
+    let count = read_count(&dbs.node, &predicate_address.contract, &key).await;
+    assert_eq!(count, new_count);
 
-    app.increment().await.unwrap();
+    let o = utils::builder::build_default(&dbs).await.unwrap();
+    assert_eq!(o.succeeded.len(), 1);
 
-    wait_for_change(&app, 1).await;
+    // FIXME: Shouldn't this be 1?
+    assert_eq!(o.failed.len(), 2);
```
Comment (supiket) [inline apps/counter/app/tests/tests.rs:57]: What is going on here?

File: apps/token/pint/token/src/contract.pnt [hunk 5]
```diff
@@ -127,21 +154,39 @@ predicate Transfer {
...
-    @auth(key; auth_addr; auth::@transfer(); auth::@transfer_with());
+    constraint match auth.mode {
+        TransferAuthMode::Signed(auth) => match auth.mode {
+            TransferSignedMode::All => @verify_key({key, to, amount, nonce'}; auth.sig; key),
+            TransferSignedMode::Key => @verify_key({key, nonce'}; auth.sig; key),
+            TransferSignedMode::KeyTo => @verify_key({key, to, nonce'}; auth.sig; key),
+            TransferSignedMode::KeyAmount => @verify_key({key, amount, nonce'}; auth.sig; key),
+        },
+        TransferAuthMode::Predicate(addr) => @check_if_predicate_owns(addr; key),
+    };
```
Comment (supiket) [inline apps/token/pint/token/src/contract.pnt:165]: This part may need verbal explanation, maybe we should mention somewhere why these exist, i.e. maybe the user wants to pay the solver but they don't know who the solver is so they don't sign over the `to`

**Review by supiket — APPROVED** [review]
Awesome

**Pull request merged**